web/README.md

```diff
@@ -0,0 +1,118 @@
+
+|Default|rules?|Action|
+|-|-|-|
+|block|no|All third-party requests from this tracker are blocked|
+|block|yes|If a rule was matched then follow the rule action, otherwise blocked|
+|ignore|yes|If a rule was matched then follow the rule action. If no action exists then block the request. Allow all other requests to load[^1]|
+|ignore|no|Do not block[^1]|
+
+### rules
+
+An optional array of objects containing regexes to match against the full URL of third-party requests made to this domain. A matching rule takes priority over the tracker default action. 
+
+|rule|Regex to match against the full tracker URL
+|:-|:-|
+|__fingerprinting__|__see [fingerprinting](#fingerprinting-0-3)__|
+|__cookies__|__see [cookies](#cookies-0-1)__|
+|__surrogate__|__Certain tracking scripts are implemented in a way that attaches function calls to page elements. When these scripts are blocked, they break sites. In order to block these trackers while still maintaining site functionality, we redirect the requests to surrogate code that replaces all of their functions with no-ops. This field contains the file name of the [replacement code](https://github.com/duckduckgo/tracker-surrogates) to serve instead of blocking__|
+|exceptions|Optional object listing types and domains to not block on|
+|action|Optional action to apply other than blocking|
+
+**exceptions**
+
+An optional object that can contain `domains` or `types` arrays. Do not block the tracker request if there is a `domains` or `types` match. In cases where the exceptions object contains both `domains` and `types` both must match. 
+
+|domains|An array of domains to match against the site requesting the tracker. Do not block in cases where the site domain matches an entry in the `domains` array|
+|:-|:-|
+|__types__|__An array of [resource types](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/ResourceType) to match against the tracker request resource type. Do not block in cases where the tracker resource type matches an entry in the `types` array__|
+
+**action**
+
+An optional field listing an action to take when matching on the the rule other than blocking. The `action` field can also be used to swtich a matched rule to a non-blocking rule by setting `action: ignore`[^1].
+
+## Entities
+
+The entities object contains an entry for each of the trackers contained in the blocklist and comes directly from the [entity data](https://github.com/duckduckgo/tracker-radar/tree/main/entities) in Tracker Radar. 
+
+## Domains
+
+The domains object is a mapping of all entity properties to entity name. This is used for quick lookup of the site owner for determining if a request is first or third-party. 
+
+## CNAMEs
+
+Used for [CNAME cloadking protection](https://help.duckduckgo.com/duckduckgo-help-pages/privacy/web-tracking-protections/#cname-cloaking-protection). The cnames object maps a first-party subdomain that has a DNS CNAME redirecting to a third-party tracker.
```

Comment:
```suggestion
Used for [CNAME cloaking protection](https://help.duckduckgo.com/duckduckgo-help-pages/privacy/web-tracking-protections/#cname-cloaking-protection). The cnames object maps a first-party subdomain that has a DNS CNAME redirecting to a third-party tracker.
```

ad king!

Reply:
I guess it's a technicality, but domain is not always first-party. As we discussed the other day if tracker.com hides behind tracker.safe.com and is loaded on bla.com (not a first party), we will still block it.